osbuild/formats/v2.py

```diff
@@ -391,9 +391,22 @@ def load(description: Dict, index: Index) -> Manifest:
 
 
 #pylint: disable=too-many-branches
-def output(manifest: Manifest, res: Dict) -> Dict:
+def output(manifest: Manifest, res: Dict, store=None) -> Dict:
     """Convert a result into the v2 format"""
 
+    def collect_metadata(p: Pipeline) -> Dict[str, Any]:
+        data: Dict[str, Any] = {}
+        r = res.get(p.id, {})
+        for stage in r.get("stages", []):
+            md = stage.metadata
+            if not md:
+                continue
+            name = stage.name
+            val = data.setdefault(name, {})
```

Comment:
(commit: `format/v2: extract metadata...`)

This could be merged, I guess? `name` is not used anywhere else.

Reply:
True, I didnt wanna change the code, but just move it, but why not.